Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml

```diff
@@ -1,58 +1,72 @@
-  | where clientInfo_s !~ "None" and isnotempty(clientInfo_s)
-  | where requestUri_s !~ "None" and isnotempty(requestUri_s)
-  | where OperationName in~ (OperationList)   
-  ) on identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g 
-  | summarize EventCount=sum(count_), StartTimeUtc=min(TimeGenerated), EndTimeUtc=max(TimeGenerated), TimeTriggered=makelist(TimeGenerated),OperationNameList=make_set(OperationName), RequestURLList=make_set(requestUri_s), CallerIPList = make_set(CallerIPAddress),  CallerIPMax= arg_max(CallerIPAddress,*) by ResourceType, ResultType, Resource, id_s, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, clientInfo_s
-  | extend timestamp = EndTimeUtc, IPCustomEntity = CallerIPMax, AccountCustomEntity = identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g
+  let AllowedAppId = dynamic(["509e4652-da8d-478d-a730-e9d4a1996ca4","8cae6e77-e04e-42ce-b5cb-50d82bce26b1"]);
+  let OperationList = dynamic(["SecretGet", "KeyGet", "VaultGet"]);
+  let Retrievals = materialize(
+      AzureDiagnostics
+      | where OperationName in (OperationList) and ResourceType =~ "VAULTS"
+      | where not(identity_claim_appid_g in (AllowedAppId) and OperationName == 'VaultGet')
+      | extend
+          ResultType = column_ifexists("ResultType", ""),
+          identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = column_ifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", ""),
+          identity_claim_http_schemas_xmlsoap_org_ws_2005_05_identity_claims_upn_s = column_ifexists("identity_claim_http_schemas_xmlsoap_org_ws_2005_05_identity_claims_upn_s", ""),
+          identity_claim_oid_g = column_ifexists("identity_claim_oid_g", ""),
+          identity_claim_upn_s = column_ifexists("identity_claim_upn_s", "")
+      | extend
+          CallerObjectId = iff(isempty(identity_claim_oid_g), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, identity_claim_oid_g),
+          CallerObjectUPN = iff(isempty(identity_claim_upn_s), identity_claim_http_schemas_xmlsoap_org_ws_2005_05_identity_claims_upn_s, identity_claim_upn_s)
+  );
+  let CallerObjectIds = materialize(
+      Retrievals
+      | where ResultType == "Success"
+      | summarize Count = count() by OperationName, CallerObjectId
+      | where Count > EventCountThreshold
+      | distinct CallerObjectId
+  );
+  Retrievals
+  | where isnotempty(toscalar(CallerObjectIds)) and CallerObjectId in (CallerObjectIds)
+  | extend
+      requestUri_s = column_ifexists("requestUri_s", ""),
+      id_s = column_ifexists("id_s", ""),
+      CallerIPAddress = column_ifexists("CallerIPAddress", ""),
+      clientInfo_s = column_ifexists("clientInfo_s", "")
+  | summarize
+      EventCount = count(),
+      StartTime = min(TimeGenerated),
+      EndTime = max(TimeGenerated),
+      //TimeTriggered = make_list(TimeGenerated, 250),
```

Comment:
can you reduce soft limit from 250 to 25/50 for better readability in UI and avoid perf issues..

Reply:
Done.